controllers/disruption_controller.go

```diff
@@ -431,6 +434,40 @@ func (r *DisruptionReconciler) cleanDisruption(instance *chaosv1beta1.Disruption
 	return cleaned, nil
 }
 
+func (r *DisruptionReconciler) handleOrphanedChaosPods(req ctrl.Request) error {
+	ls := make(map[string]string)
+
+	ls[chaostypes.DisruptionNameLabel] = req.Name
+	ls[chaostypes.DisruptionNamespaceLabel] = req.Namespace
+
+	chaosPods, err := r.getChaosPods(nil, ls)
+	if err != nil {
+		return err
+	}
+
+	for _, chaosPod := range chaosPods {
+		r.handleMetricSinkError(r.MetricsSink.MetricOrphanFound([]string{"name:" + req.Name, "namespace:" + req.Namespace}))
+
+		r.log.Warnw("orphaned chaos pod detected, will attempt to delete", "chaosPod", chaosPod.Name)
+		controllerutil.RemoveFinalizer(&chaosPod, chaostypes.ChaosPodFinalizer)
+
+		if err := r.Client.Update(context.Background(), &chaosPod); err != nil {
+			r.log.Errorw("error removing chaos pod finalizer", "error", err, "chaosPod", chaosPod.Name)
+
+			continue
+		}
+
+		if err := r.Client.Delete(context.Background(), &chaosPod); err != nil {
+			r.log.Errorw("error deleting orphaned chaos pod", "error", err, "chaosPod", chaosPod.Name)
+
+			continue
+		}
+		r.handleMetricSinkError(r.MetricsSink.MetricOrphanDeleted([]string{"name:" + req.Name, "namespace:" + req.Namespace}))
```

Comment:
Right now we're incrementing one metric for each found orphan, and one for each deleted orphan. We could switch to one increment every time we reconcile loop for orphans (rather than for each orphan)? Also possible to increment for every attempted delete, rather than every successful

Reply:
I'd either go with a metric for each found orphan or a metric for each call to this function if orphans are found (whatever the amount of orphans).

Reply:
Are you suggesting only one metric total, (in which case I'd prefer one per found orphan), and thus no metric for deleted orphans?

Reply:
Yup exactly